R/get-param.R

```diff
@@ -0,0 +1,128 @@
+#' Return theta values as a named vector
+#'
+#' @param .summary  A `bbi_nonmem_summary` or `bbi_summary_list` object.
+#' @export
+get_theta <- function(.summary){
+
+  if (inherits(.summary, "bbi_nonmem_summary")) {
+    .summary <- list(.summary)
+    check_model_object_list(.summary, .mod_types = c(NM_SUM_CLASS))
+  }else if(inherits(.summary, "bbi_summary_list")){
+    check_model_object(.summary, .mod_types = c("bbi_summary_list"))
+  }
+
+  thetas <- map(.summary, ~ {
+    # unpack bbi_summary_list element
+    if (!is.null(.x$bbi_summary)) .x <- .x$bbi_summary
+    param_names <- .x[[SUMMARY_PARAM_NAMES]]
+    theta <- .x[[SUMMARY_PARAM_DATA]][[length(.x[[SUMMARY_PARAM_DATA]])]]$estimates$theta
+    names(theta) <- param_names$theta
+    theta
+  })
+
+
+  if(length(thetas) == 1) thetas <- thetas[[1]]
+
+  return(thetas)
+}
+
+
+#' Return omega values as a matrix
+#'
+#' @param .summary  A `bbi_nonmem_summary` or `bbi_summary_list` object.
+#' @export
+get_omega <- function(.summary){
+
+  if (inherits(.summary, "bbi_nonmem_summary")) {
+    .summary <- list(.summary)
+    check_model_object_list(.summary, .mod_types = c(NM_SUM_CLASS))
+  }else if(inherits(.summary, "bbi_summary_list")){
+    check_model_object(.summary, .mod_types = c("bbi_summary_list"))
+  }
+
+  omegas <- map(.summary, ~ {
+    # unpack bbi_summary_list element
+    if (!is.null(.x$bbi_summary)) .x <- .x$bbi_summary
+
+    param_names <- .x[[SUMMARY_PARAM_NAMES]]
+    omega <- with(
+      .x[[SUMMARY_PARAM_DATA]][[length(.x[[SUMMARY_PARAM_DATA]])]],
+      format_matrix(estimates$omega, param_names$omega, .type = "OMEGA")
+    )
+    omega
+  })
+
+
+  if(length(omegas) == 1) omegas <- omegas[[1]]
+
+  return(omegas)
```

Comment:
@kyleam Im now wondering if we want to export this other function, rather than making it internal. I currently have something like this:

![image](https://user-images.githubusercontent.com/18685116/179794522-214a40d4-5d5f-4c42-a07e-1849ea571799.png)

KyleB was [suggesting](https://github.com/metrumresearchgroup/bbr/issues/442#issuecomment-976746226) something like this (though return everything)

Reply:
Hmm, but given it doesn't return everything, I'm not sure what functionality it adds to the interface, so my initial reaction is that it'd be better to leave it internal.  (And, when in doubt, I think internal is a good default because it's straightforward to expose something in a later release, but it's breaking to go in the other direction.)

Reply:
@kyleam I actually just refactored it again to be more in line with what Kyle was suggesting, so it now has some value, and returns an object similar to how `model_summaries()` does:

### Single summary
```r
> SUM1 %>% get_param()
$omega
          OMEGA_1  OMEGA_2
OMEGA_1 0.0985328 0.000000
OMEGA_2 0.0000000 0.156825

$sigma
        SIGMA_1
SIGMA_1       1

$theta
     THETA1      THETA2      THETA3      THETA4      THETA5 
  2.3171600  54.6151000 462.5140000  -0.0820082   4.1795900
```

### summary list
```r
> sums %>% get_param()
$omega
$omega[[1]]
          OMEGA_1  OMEGA_2
OMEGA_1 0.0985328 0.000000
OMEGA_2 0.0000000 0.156825

$omega[[2]]
             OMEGA_1     OMEGA_2     OMEGA_3      OMEGA_4
OMEGA_1  0.010410500 0.000319542 0.001977200 -0.000189864
OMEGA_2  0.000319542 0.008278020 0.000872168  0.001005720
OMEGA_3  0.001977200 0.000872168 0.013139700  0.004040370
OMEGA_4 -0.000189864 0.001005720 0.004040370  0.010987300


$sigma
$sigma[[1]]
        SIGMA_1
SIGMA_1       1

$sigma[[2]]
        SIGMA_1
SIGMA_1       1


$theta
$theta[[1]]
     THETA1      THETA2      THETA3      THETA4      THETA5 
  2.3171600  54.6151000 462.5140000  -0.0820082   4.1795900 

$theta[[2]]
    THETA1     THETA2     THETA3     THETA4     THETA5     THETA6     THETA7     THETA8     THETA9    THETA10    THETA11 
 3.3005000  3.2519700 -0.6122190 -0.2080120  0.7372940  1.1359900  0.3346290  0.1923520  0.6916970  2.2993700  0.0991684 
```

Reply:
> so it now has some value

That's true, though I'm not really a fan.  I think we should choose one approach, not expose a function that returns all of the parameters _and_ accessor functions for the individual parameters.  It complicates the interface and code.  

That's of course just one opinion.  Perhaps others disagree with me here.

Reply:
The main thing I like is being _**able to**_ retrieve all parameters, but not necessarily forced to. Perhaps I could make `"omega"` the default instead of all 3. If we chose to keep the 3 specific functions (and hide `get_param()`, I think I would still leave the ability to map across all 3 parameters just in case we want that functionality in the future.

@callistosp @kylebaron what do you guys think of the above change (see code snippets)? Does it make sense to have 1 general function, a 3 specific functions in terms of how you would/could use them?

Reply:
I think I agree with @kyleam to err on the side of keeping it simple here (i.e. _don't_ add the extra helper). Also, I think Sam already weighed in on this [on the issue](https://github.com/metrumresearchgroup/bbr/issues/442#issuecomment-1180619733) and seemed to agree that the helper wouldn't likely be necessary in most cases.